### ISSUE TITLE
SBIED tutorial does not always reflect best practices for performance

### ISSUE THREAD

**Issue by AJAbkemeier** [pypomp/tutorials#6]
There are some places where the SBIED tutorial uses random variate generators from `jax.random` instead of `pypomp.random`. The JAX versions are very slow on GPUs. In addition, the code sometimes calls these generators multiple times in a row instead of taking advantage of their inherent vectorization. This should be fixed to be in line with the [best practices](https://pypomp.readthedocs.io/en/latest/best_practices.html) for performance in pypomp. 

Minor detail: there are some places where e.g. `jnp.float32()` is used where it is seemingly not necessary. I would suggest removing these so as not to confuse new users.

**Comment by ionides** [comment]
I can do this soon. I wonder if @kunyanghe21 has opinions on the `jnp.float32` issue. I agree these are best avoided unless necessary (which they shouldn't be). If they are necessary for some reason, that should be explained.
